### PR TITLE
Add more minor enhancements to ItaniumDemangle.h.

### PR DESCRIPTION
**Pull request by kbobrovs**
- Add accessor method for the ObjCProtoName::Protocol field so that
  other Nodes such as PointerType can access it.
- Make TemplateArgs::getParams() const and return reference instead of
  a copy.

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

**Comment by kbobrovs** [comment]
Need more changes in demangler to get LowerESIMD.cpp compiled

File: llvm/include/llvm/Demangle/ItaniumDemangle.h [hunk 1]
```diff
@@ -497,7 +499,7 @@ class PointerType final : public Node {
     } else {
       const auto *objcProto = static_cast<const ObjCProtoName *>(Pointee);
       s += "id<";
-      s += objcProto->Protocol;
+      s += objcProto->getProtocol();
```
Comment (bader) [inline llvm/include/llvm/Demangle/ItaniumDemangle.h:502]: Is this change necessary?

Comment (kbobrovs) [reply llvm/include/llvm/Demangle/ItaniumDemangle.h:502]: Yes. Protocol is private field accessed in the subclass.

Comment (bader) [reply llvm/include/llvm/Demangle/ItaniumDemangle.h:502]: `PointerType` is `ObjCProtoName ` friend, so it has access to private members.

Comment (kbobrovs) [reply llvm/include/llvm/Demangle/ItaniumDemangle.h:502]: Good point. I had compilation failure for the `objcProto->Protocol` before. Now after I revert this change compilation still goes fine. Will remove the change,
